src/components/chat/chat.js

```diff
@@ -44,15 +56,22 @@ class Chat extends React.Component {
       lineNumbers: true,
       content: this.textArea.current,
     })
+    this.codemirror.setSize(null, '80vh');
     this.codemirror.on('blur', this.codeChanged);
     this.props.socket.on('editor', (data) => this.codemirror.getDoc().setValue(data.code)); // handles received text
     this.props.socket.on('newUser', this.updateCode); // this code is not working - what was its purpose?
-  }
 
-  shouldComponentUpdate() {
-    return false;
+    //HANG-UP
+    this.props.socket.on('hang up', () => {
+      this.openChatModal()
+    })
   }
 
+  //WHY WAS THIS HERE? -- This was added by Arol to prevent the editor from continuously re-rendering
+  // shouldComponentUpdate() {
+  //   return false;
+  // }
```

Comment:
Should we remove this since we need it to re-render? We might move the editor to a separate component so it doesn't re-render as we were having issues with.